### PR TITLE
Feature/sre 593/openstack blockstorage volume v3

### PR DESCRIPTION
**Pull request by Shonei**

### CODE REVIEW

File: openstack/data_source_openstack_blockstorage_volume_v3.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
+package openstack
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/v3/volumes"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
```
Comment (dimitarvdimitrov) [inline openstack/data_source_openstack_blockstorage_volume_v3.go:9]: Is this the convention that tf-openstack are using?

**Review by dimitarvdimitrov — APPROVED** [review]

File: openstack/data_source_openstack_blockstorage_volume_v3.go (hunk 1 above)
Comment (Shonei) [reply openstack/data_source_openstack_blockstorage_volume_v3.go:9]: That is how the order appears in other files.

File: openstack/data_source_openstack_blockstorage_volume_v3_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	return volume.ID, nil
+}
+
+func testAccBlockStorageV3DeleteVolume(t *testing.T, volumeID string) {
+	config, err := testAccAuthFromEnv()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	bsClient, err := config.BlockStorageV2Client(OS_REGION_NAME)
```
Comment (comradedraganov) [inline openstack/data_source_openstack_blockstorage_volume_v3_test.go]: Should be V3

File: openstack/data_source_openstack_blockstorage_volume_v3_test.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+		}
+
+		return nil
+	}
+}
+
+func testAccBlockStorageV3VolumeDataSource_basic(snapshotName string) string {
+	return fmt.Sprintf(`
+    data "openstack_blockstorage_volume_v3" "volume_1" {
+      name        = "%s"
```
Comment (comradedraganov) [inline openstack/data_source_openstack_blockstorage_volume_v3_test.go]: Why is there a bunch of spacing here?

File: website/docs/d/blockstorage_volume_v3.html.markdown [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+```hcl
+data "openstack_blockstorage_volume_v3" "volume_1" {
+  name = "volume_1"
+}
+```
+
+## Argument Reference
+
+* `region` - (Optional) The region in which to obtain the V2 Block Storage
```
Comment (comradedraganov) [inline website/docs/d/blockstorage_volume_v3.html.markdown]: V3

File: website/openstack.erb [hunk 5]
```diff
@@ -25,6 +25,9 @@
             <li<%= sidebar_current("docs-openstack-datasource-blockstorage-volume-v2") %>>
               <a href="/docs/providers/openstack/d/blockstorage_volume_v2.html">openstack_blockstorage_volume_v2</a>
             </li>
+            <li<%= sidebar_current("docs-openstack-datasource-blockstorage-volume-v3") %>>
+              <a href="/docs/providers/openstack/d/blockstorage_volume_v3.html">openstack_blockstorage_volume_v2</a>
```
Comment (comradedraganov) [inline website/openstack.erb]: V3

File: openstack/data_source_openstack_blockstorage_volume_v3_test.go (hunk 3 above)
Comment (Shonei) [reply openstack/data_source_openstack_blockstorage_volume_v3_test.go]: It is how it was copied. There shouldn't be. It is removed now.

**Review by comradedraganov — APPROVED** [review]

**Pull request merged**